tools/vagrant-box-atom/upload.rb

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/ruby
+
+require 'bundler/setup'
+require 'net/http'
+require 'vagrant_cloud'
+
+ORG = 'artefactual'
+BOX = 'atom'
+PROVIDER = 'virtualbox'
+
+path = ARGV[0]
+token = ARGV[1]
+version = ARGV[2]
+description = ARGV[3]
+
+client = VagrantCloud::Client.new(access_token: token)
+
+client.box_version_create(
+    username: ORG,
+    name: BOX,
+    version: version,
+    description: description
+)
+
+client.box_version_provider_create(
+    username: ORG,
+    name: BOX,
+    version: version,
+    provider: PROVIDER
+)
+
+upload_url = client.box_version_provider_upload(
+    username: ORG,
+    name: BOX,
+    version: version,
+    provider: PROVIDER
+)
```

Comment:
If anything fails here (authentication, version already exists, etc.) the script ends with error. However, this may leave residual data in Vagrant Cloud that won't be considered in a a new run. For example, if the box fails to upload below and the workflow is triggered again with the same version it will fail because the version already exists.

We could consider those cases in the script but I'm not sure if we should, as it may lead to versions overwrite. We could document the situation so, in case of failure, if the version has been created it has to be deleted from Vagrant Cloud's GUI before trying again with the same version.

Reply:
I guess we could check if the version is fully released to avoid that overwrite, but I still have my doubts about if it's worth it, feedback is welcomed.

Reply:
Maybe a workflow job with a [status condition](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions) that does the necessary clean-up?

Reply:
Thanks @sevein, I think it may be easier directly in the script, but it is an enhancement that could be added later on. For now I'll add a note in the internal wiki about failure/retry.